synapse/http/matrixfederationclient.py

```diff
@@ -473,7 +479,8 @@ def put_json(self, destination, path, args={}, data={},
                  json_data_callback=None,
                  long_retries=False, timeout=None,
                  ignore_backoff=False,
-                 backoff_on_404=False):
+                 backoff_on_404=False,
+                 try_trailing_slash_on_404=False):
```

Comment:
```suggestion
                 try_trailing_slash_on_404=False,
             ):
```

Reply:
flake unfortunately complains about visual indentations if we do this.

Reply:
/me shakes fist